src/gui/editorwidgets/qgshtmlwidgetwrapper.cpp

```diff
@@ -82,9 +82,13 @@ void QgsHtmlWidgetWrapper::setHtmlContext( )
   if ( !mWidget )
     return;
 
-  QgsAttributeEditorContext attributecontext = context();
+  const QgsAttributeEditorContext attributecontext = context();
   QgsExpressionContext expressionContext = layer()->createExpressionContext();
   expressionContext << QgsExpressionContextUtils::formScope( mFeature, attributecontext.attributeFormModeString() );
+  if ( attributecontext.parentFormFeature().isValid() )
```

Comment:
Hi @elpaso 

Just stumbled upon this pull request.
I'm wondering how this works for the HTML widget. As far as I can tell it is currently the responsibility for each widget implementation to keep the `parentFormFeature` in the context up to date (e.g. https://github.com/qgis/QGIS/pull/34827/files#diff-38bcb929a48491604891779e4ea4dc0dR92-R97) and the HTML widget does not provide this code. Which would mean, the feature here is always invalid? I didn't test it, so I might have missed something.

I'm tempted to discuss an API change proposal here, but I'd first like to be sure that my interpretation of this is correct.

Reply:
Nevermind, I indeed missed something. Sorry for the noise.